PayloadPkg/OsLoader/OsLoader.c

```diff
@@ -64,22 +64,21 @@ UpdateLoadedImage (
   COMMON_IMAGE               *CommonImage;
   PLATFORM_SERVICE           *PlatformService;
   CHAR8                      *TypeStr;
+  CHAR8                      BlobName[5];               // 4 character component name + null termination
+  CHAR8                      BlobAddr[17];              // 64-bit address in ASCII hex + null termination
```

Comment:
Can you use UINTN? This will automatically type case to UINT32 or UINT64 based on the build. You can still get the size using sizeof operator.